src/WebJobs.Extensions.DurableTask/Listener/OutOfProcOrchestrationShim.cs

```diff
@@ -71,33 +71,9 @@ public async Task HandleDurableTaskReplay(OrchestrationInvocationResult executio
             }
         }
 
-        internal JObject ParseOOProcResult(OrchestrationInvocationResult result)
-        {
-            JObject jObj = result.ReturnValue as JObject;
-            if (jObj == null && result.ReturnValue is string jsonText)
-            {
-                try
-                {
-                    jObj = JObject.Parse(jsonText);
-                }
-                catch
-                {
-                    throw new ArgumentException("Out of proc orchestrators must return a valid JSON schema");
-                }
-            }
-
-            if (jObj == null)
-            {
-                throw new ArgumentException("The data returned by the out-of-process function execution was not valid json.");
-            }
-
-            return jObj;
-        }
-
         internal async Task<bool> ScheduleDurableTaskEvents(OrchestrationInvocationResult result)
         {
-            JObject jObj = this.ParseOOProcResult(result);
-            var execution = JsonConvert.DeserializeObject<OutOfProcOrchestratorState>(jObj.ToString());
+            var execution = JsonConvert.DeserializeObject<OutOfProcOrchestratorState>(result.JsonString);
```

Comment:
Since we've already spent CPU cycles converting a string into a `JObject`, could we replace this redundant deserialization with a simpler call that converts the `JObject` directly into an `OutOfProcOrchestratorState`? For example:

```suggestion
            var execution = result.Json.ToObject<OutOfProcOrchestratorState>();
```

This, of course, assumes that `OrchestrationInvocationResult` has a `Json` property which is of type `JObject`.